engine/access/state_stream/filter.go

```diff
@@ -156,23 +156,23 @@ func (f *EventFilter) Match(event flow.Event) bool {
 
 // matchFieldFilter checks if the given event matches the specified field filters.
 // It returns true if the event matches any of the provided field filters, otherwise false.
-func (f *EventFilter) matchFieldFilter(event *flow.Event, fieldFilters FieldFilter) bool {
```

Comment:
why change this to pass by value? This code will get run for every event in every block for each stream. Passing by value was intended to avoid unnecessary allocations

Reply:
Ah, sorry! I saw this and my first thought was that it might have been an accident, then that it was maybe done for allowing mutation, which didn't seem to be the case, so I removed it. 

The type only consists of a pointer and a slice, so fairly trivial to pass around by value (copies are cheap), but I'll add it back

Reply:
Just seeing now this is a `flow.Event`, not a `cadence.Event` 👍

Reply:
Reverted in 0ee46de35cba20a4da38168eb209b5b763d8ce2c

Reply:
@peterargue Does this address your concerns?

Reply:
yes. approved!